src/match/views.py

```diff
@@ -0,0 +1,122 @@
+import json
+
+from api import settings as api_settings
+from api.utils import failure_response
+from api.utils import success_response
+from django.db.models import Q
+from match import match_status
+from match.models import Match
+from match.serializers import BothUsersMatchSerializer
+from match.serializers import MatchSerializer
+from rest_framework import generics
+
+from .controllers.create_match_controller import CreateMatchController
+from .controllers.update_match_controller import UpdateMatchController
+
+
+class MatchesView(generics.GenericAPIView):
+    serializer_class = BothUsersMatchSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request):
+        """Get all matches."""
+        matches = Match.objects.all().order_by("-created_date")
+        return success_response(self.serializer_class(matches, many=True).data)
+
+    def post(self, request):
+        """Create a match."""
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            data = request.data
+        return CreateMatchController(data, self.serializer_class).process()
+
+
+class MatchView(generics.GenericAPIView):
+    serializer_class = BothUsersMatchSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request, id):
+        """Get match by id."""
+        match = Match.objects.filter(id=id)
+        if match:
+            return success_response(self.serializer_class(match[0]).data)
+        return failure_response("Match does not exist")
```

Comment:
bruh just return the match lol

Reply:
also just do `Match.objects.get(id=id)` 
if it fails, it fails

Reply:
wouldn't we wanna return a 404 if client is trying to get a match that doesn't exist? that's what we do for all our other models

Reply:
hmmmmmmmmm i guess 

lol you can keep this then 
i just being a picky boi

Reply:
little nitpick, actually put the 404 error here even though it's a default argument to be explicit like in my refactor (same for success)